### PR TITLE
Kick/disconnect improvements

### PR DESCRIPTION
**Pull request by MrIvanPlays**
After some tests I've saw that the delayed close isn't necessary in ServerConnection and UserConnection, so this is why I am making this patch.

- Totally get rid of delayed close by using normal close in ServerConnection and UserConnection, built in the close delay in InitialHandler

Makes server switch speeds faster.
Closes https://github.com/SpigotMC/BungeeCord/issues/2713

### CODE REVIEW

File: proxy/src/main/java/net/md_5/bungee/connection/InitialHandler.java [hunk 1]
```diff
@@ -562,9 +563,17 @@ public void disconnect(String reason)
     @Override
     public void disconnect(final BaseComponent... reason)
     {
-        if ( thisState != State.STATUS && thisState != State.PING )
+        if ( thisState != State.STATUS && thisState != State.PING && thisState != State.HANDSHAKE )
         {
-            ch.delayedClose( new Kick( ComponentSerializer.toString( reason ) ) );
```
Comment (Janmm14) [inline proxy/src/main/java/net/md_5/bungee/connection/InitialHandler.java]: I suggest readding the comment https://github.com/SpigotMC/BungeeCord/pull/2722/files#diff-d81a4f22970536a8f3cfe44bfd87df6eL106-L108

Comment (MrIvanPlays) [reply proxy/src/main/java/net/md_5/bungee/connection/InitialHandler.java]: I suggest reading the code after that line

File: proxy/src/main/java/net/md_5/bungee/connection/InitialHandler.java [hunk 2]
```diff
@@ -562,9 +563,17 @@ public void disconnect(String reason)
...
+            ch.getHandle().eventLoop().schedule( new Runnable()
```
Comment (Janmm14) [inline proxy/src/main/java/net/md_5/bungee/connection/InitialHandler.java]:
```suggestion
            // Minecraft client can take some time to switch protocols.
            // Sending the wrong disconnect packet whilst a protocol switch is in progress will crash it.
            // Delay 250ms to ensure that the protocol switch (if any) has definitely taken place.
            ch.getHandle().eventLoop().schedule( new Runnable()
```

**Comment by MrIvanPlays** [comment]
@md-5 can I get any response on this please?

**Comment by md-5** [comment]
PR _may_ be valid from a general design/quality perspective but I dispute the claim that it 'Makes server switch speeds faster.' or achieves anything else really.

The unnecessary moving of code dropped use of the `closing` variable which could introduce issues -- this should be a 2 line PR anyway.

**Comment by MrIvanPlays** [comment]
There. 2 line PR. Now will you merge or no?

**Pull request closed without merging**